drivers/flash/flash_stm32h7x.c

```diff
@@ -112,11 +170,161 @@ uint8_t flash_stm32_get_rdp_level(const struct device *dev)
 
 void flash_stm32_set_rdp_level(const struct device *dev, uint8_t level)
 {
-	write_optb(dev, FLASH_OPTSR_RDP_Msk,
-		(uint32_t)level << FLASH_OPTSR_RDP_Pos);
+	write_optb(dev, FLASH_OPTSR_RDP_Msk, (uint32_t)level << FLASH_OPTSR_RDP_Pos);
 }
 #endif /* CONFIG_FLASH_STM32_READOUT_PROTECTION */
 
+#if defined(CONFIG_FLASH_STM32_WRITE_PROTECT)
+
+int flash_stm32_update_wp_sectors(const struct device *dev, uint64_t changed_sectors,
+				  uint64_t protected_sectors)
+{
+#ifdef CONFIG_SOC_SERIES_STM32H7RSX
+	uint64_t bank_mask = FLASH_WRPSRP_WRPS_Msk >> FLASH_WRPSRP_WRPS_Pos;
+	uint32_t pos = FLASH_WRPSRP_WRPS_Pos;
+#else
+	uint64_t bank_mask = FLASH_WPSN_WRPSN_Msk >> FLASH_WPSN_WRPSN_Pos;
+	uint32_t pos = FLASH_WPSN_WRPSN_Pos;
+#endif
+	uint32_t sectors_per_bank = 32 - __builtin_clz(bank_mask);
```

Comment:
Do we have a guarantee that each bank has the same number of sectors?

Reply:
Yes, from I saw in the HAL, there is always one mask defined for both banks.